static/app/components/smartSearchBar/index.tsx

```diff
@@ -923,17 +924,17 @@ class SmartSearchBar extends Component<Props, State> {
   /**
    * Returns array of possible key values that substring match `query`
    */
-  getTagKeys(query: string): [SearchItem[], ItemType] {
+  getTagKeys(searchTerm: string): [SearchItem[], ItemType] {
     const {prepareQuery, supportedTagType} = this.props;
 
     const supportedTags = this.props.supportedTags ?? {};
 
-    let tagKeys = Object.keys(supportedTags);
+    let tagKeys = Object.keys(supportedTags).sort((a, b) => a.localeCompare(b));
```

Comment:
Sort the tag keys here instead of inside `getTagItemsFromKeys` so they go into `filterKeysFromQuery` already sorted